### PR TITLE
Add `DateTime` trigger

### LINKED ISSUE #153: [triggers] Add the new `Time is <Item>` trigger
Add support for the DateTime Trigger (https://github.com/openhab/openhab-core/issues/2726).

See https://github.com/openhab/openhab-core/pull/2963 for implementation in Rules DSL (might help).

### PR DESCRIPTION
**Pull request by florian-h05**
Fixes #153.

This adds support for the DateTime trigger, which was added in
https://github.com/openhab/openhab-core/issues/2726.

### CODE REVIEW

File: rules/rules.js [hunk 1]
```diff
@@ -22,8 +23,8 @@
  * @property {string} oldStatus only for {@link triggers.ThingStatusChangeTrigger}: Previous state of Thing that triggered event
  * @property {string} newStatus only for {@link triggers.ThingStatusChangeTrigger}: New state of Thing that triggered event
  * @property {string} status only for {@link triggers.ThingStatusUpdateTrigger}: State of Thing that triggered event
- * @property {string} eventType for all triggers except {@link triggers.PWMTrigger}, {@link triggers.PIDTrigger}: Type of event that triggered event (change, command, time, triggered, update)
- * @property {string} triggerType for all triggers except {@link triggers.PWMTrigger}, {@link triggers.PIDTrigger}: Type of trigger that triggered event (for `TimeOfDayTrigger`: `GenericCronTrigger`)
+ * @property {string} eventType for all triggers except {@link triggers.PWMTrigger}, {@link triggers.PIDTrigger}, time triggers: Type of event that triggered event (change, command, time, triggered, update)
```
Comment (digitaldan) [inline rules/rules.js:26]: when we say "time triggers", do we mean `TimeOfDayTrigger` and `DateTimeTrigger`?  If so, should we just call those out directly ?

**Review by digitaldan — APPROVED** [review]
LGTM, thanks!  Only one small question on docs, but other then that looks good to merge.

File: rules/rules.js (hunk 1 above)
Comment (florian-h05) [reply rules/rules.js:26]: We mean all three time triggers that we have:
- `TimeOfDayTrigger`
- `GenericCronTrigger`
- `DateTimeTrigger`

**Comment by florian-h05** [comment]
@digitaldan 
Can I merge?

**Comment by digitaldan** [comment]
sorry, yes i'm fine, thanks!

**Pull request merged**